tls/s2n_signature_algorithms.c

```diff
@@ -179,11 +175,9 @@ int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn,
 
 int s2n_send_supported_sig_scheme_list(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
-    const struct s2n_security_policy *security_policy = NULL;
-    GUARD(s2n_connection_get_security_policy(conn, &security_policy));
-    notnull_check(security_policy);
-
-    const struct s2n_signature_preferences *signature_preferences = security_policy->signature_preferences;
+    notnull_check(conn);
```

Comment:
Is this notnull necessary?

Reply:
It is de-referenced within the `s2n_connection_get_signature_preferences` call. We also do a notnull_check within that function for conn, so by adding this check we are catching the error earlier. I can get rid of it.